src/Persistence/Yaml/YamlSerializer.cs

```diff
@@ -19,21 +20,57 @@ public string Serialize(object graph)
     {
         _ = graph ?? throw new ArgumentNullException(nameof(graph));
 
-        return _serializer.Serialize(graph);
+        return SerializeCore(graph);
     }
 
     public string SerializeControl<T>(T graph) where T : Control
     {
         _ = graph ?? throw new ArgumentNullException(nameof(graph));
 
-        return _serializer.Serialize(graph);
+        return SerializeCore(graph);
     }
 
     public void SerializeControl<T>(TextWriter writer, T graph) where T : Control
     {
         _ = writer ?? throw new ArgumentNullException(nameof(writer));
         _ = graph ?? throw new ArgumentNullException(nameof(graph));
 
-        _serializer.Serialize(writer, graph);
+        SerializeCore(writer, graph);
+    }
+
+    private void SerializeCore<T>(TextWriter writer, T graph)
+    {
+        try
+        {
+            _serializer.Serialize(writer, graph);
+        }
+        catch (YamlException ex)
+        {
+            throw ex.Start.Equals(Mark.Empty)
+                ? new PersistenceException(PersistenceErrorCode.SerializationError, ex.Message, ex)
+                : new PersistenceException(PersistenceErrorCode.SerializationError, ex.Message, ex)
+                {
+                    LineNumber = ex.Start.Line,
+                    Column = ex.Start.Column,
+                };
+        }
+    }
+
+    private string SerializeCore<T>(T graph)
+    {
+        try
+        {
+            return _serializer.Serialize(graph);
+        }
+        catch (YamlException ex)
```

Comment:
Should we catch all instead of just YamlException?

Reply:
no. Please see: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1031